pkg/reconciler/gc/v1/gc_test.go

```diff
@@ -165,41 +162,66 @@ func TestGCReconcile(t *testing.T) {
 				WithCreationTimestamp(old),
 				WithLastPinned(tenMinutesAgo)),
 		},
-		Key: "foo/keep-two",
 	}, {
-		Name: "keep stale revision because of minimum generations",
-		Objects: []runtime.Object{
-			cfg("keep-all", "foo", 5554,
-				// Don't set the latest ready revision here
-				// since those by default are always retained
-				WithLatestCreated("keep-all"),
-				WithConfigObservedGen),
+		name: "keep stale revision because of minimum generations",
+		cfg: cfg("keep-all", "foo", 5554,
+			// Don't set the latest ready revision here
+			// since those by default are always retained
+			WithLatestCreated("keep-all"),
+			WithConfigObservedGen),
+		revs: []*v1.Revision{
 			rev("keep-all", "foo", 5554,
 				WithRevName("keep-all"),
 				WithCreationTimestamp(oldest),
 				WithLastPinned(tenMinutesAgo)),
 		},
-		Key: "foo/keep-all",
 	}}
 
-	table.Test(t, MakeFactory(func(ctx context.Context, listers *Listers, cmw configmap.Watcher) controller.Reconciler {
-		r := &reconciler{
-			client:         servingclient.Get(ctx),
-			revisionLister: listers.GetRevisionLister(),
-		}
-		return configreconciler.NewReconciler(ctx, logging.FromContext(ctx),
-			servingclient.Get(ctx), listers.GetConfigurationLister(),
-			controller.GetEventRecorder(ctx), r, controller.Options{
-				ConfigStore: &testConfigStore{
-					config: &config.Config{
-						RevisionGC: &gcconfig.Config{
-							StaleRevisionCreateDelay:        5 * time.Minute,
-							StaleRevisionTimeout:            5 * time.Minute,
-							StaleRevisionMinimumGenerations: 2,
-						},
-					},
-				}})
-	}))
+	cfgMap := &config.Config{
+		RevisionGC: &gcconfig.Config{
+			StaleRevisionCreateDelay:        5 * time.Minute,
+			StaleRevisionTimeout:            5 * time.Minute,
+			StaleRevisionMinimumGenerations: 2,
+		},
+	}
+
+	for _, test := range table {
+		t.Run(test.name, func(t *testing.T) {
+			ctx, _ := SetupFakeContext(t)
+			ctx = config.ToContext(ctx, cfgMap)
+			client := fakeservingclient.Get(ctx)
+
+			ri := fakerevisioninformer.Get(ctx)
+			for _, rev := range test.revs {
+				ri.Informer().GetIndexer().Add(rev)
+			}
+
+			recorderList := ActionRecorderList{client}
+
+			Collect(ctx, client, ri.Lister(), test.cfg)
+
+			actions, err := recorderList.ActionsByVerb()
+			if err != nil {
+				t.Errorf("Error capturing actions by verb: %q", err)
```

Comment:
```suggestion
				t.Errorf("Error capturing actions by verb: %v", err)
```

Reply:
I was going to leave this as is to match: https://github.com/knative/serving/blob/47cd0d7fc7cbcfec9de1d17263bf58457c3d283d/vendor/knative.dev/pkg/reconciler/testing/table.go#L156

the errors we get elsewhere in this case